### DISCUSSION TITLE
label_values() should warn if a metric returns more than one timeseries per label

### DISCUSSION THREAD

**Discussion by discordianfish** [grafana/grafana#39137] [Dashboards, Panels & Visualizations]
The templating function label_values() takes a query and extracts the values for the given label. Using an expression that returns more than one timeseries for the given label/key combination is wasteful. Yet it's being done quite often, for instance in the official kubernetes mixins.

This could be avoided if label_values() would print a warning in that case.

This proposes that given a metric like this:
```
foo{label="value",bar="baz"}
foo{label="value",bar="bux"}
```

And a templating var query like this: `label_values(foo, label)` should warn that the for each label value there were two timeseries returned.

**Comment by roidelapluie** [comment]
Mmh I use this a lot and it's a feature to me. I frequently use, e.g. `label_values(up{job="node"}, env)` then in another variable reuse it: `label_values(up{job="node",env="$env"}, instance)`

**Reply by discordianfish** [reply]
Yeah sometimes it's also not avoidable. But if even official ones like the kubernetes-mixins use excessively expensive queries, it's probably worth warning the user somehow. Maybe it should warn if the expression returns *many more than one* timeseries?

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). Due to the age and number of responses to this discussion, we are deciding to close it. If this is something you would like to see in Grafana, feel free to open an [issue](https://github.com/grafana/grafana/issues) so the discussion can continue. Thank you!

**Comment by discordianfish** [comment]
@gelicia I know about the issue, so it doesn't hurt me but your users that don't know about this. So if you care about the problem, move this to an issue

**Reply by gelicia** [reply]
#82911
